### PR TITLE
fix: move position_ids fallback inside make_cp_batch_and_ctx

### PR DESCRIPTION
**Pull request by akoumpa**
Changes:

1. move the following
```
    if (
        "position_ids" not in batch
        and "input_ids" in batch
        and device_mesh is not None
        and (device_mesh["context_parallel"].size() > 1 or device_mesh["tensor_parallel"].size() > 1)
    ):
        batch["position_ids"] = torch.arange(0, batch["input_ids"].shape[1]).unsqueeze(0).to(batch["input_ids"].device)
```
inside the `make_cp_batch_and_ctx` function.

2. rename `mask` to `loss_mask` to differentiate from `attention_mask`.

### CODE REVIEW

**Review by HuiyingLi — APPROVED** [review]
LGTM
